src/backend/UserDb.js

```diff
@@ -12,25 +12,33 @@ class UserDb {
 
     async userLogin(un, pw) {
         const doc = await this.programDb.find({ username: un, password: pw });
-        return doc.length !== 0;
+        if (doc.some(item => item.enabled === 1)) {
```

Comment:
doc should be an array of users if I understand correctly. To be more accurate, you should check if the array is length 1 (which it should be) and then get the first element. Not iterate over all of them and check. At the very least you should assert that ALL of them match using every since RN if this returned every user by some error then it could pass.

Reply:
As I understood it, doc is the result of the query, which in this case is one user. Maybe I am wrong?

Reply:
.find will find all users matching username=un and password=pw. This is a list. You then iterate this list (using .some) and check if any of them are enabled.

Reply:
I see what you mean, but we also don’t allow for duplicates in the db when adding them so I assume this wouldn’t happen, but I can see where it can go wrong

Reply:
Not sure why this one was resolved